### DISCUSSION TITLE
How to update many related records?

### DISCUSSION THREAD

**Discussion by moroshko** [prisma/prisma#7547] [Q&A] [answered]
Assume that `User` can have many `Post`s:

```prisma
model User {
  id    String @id
  name  String
  posts Post[]
}

model Post {
  id     String @id
  text   String
  userId String
  user   User   @relation(fields: [userId], references: [id])
}
```

Let's say that the database already contains users and posts, and I get a new (updated) set of users and posts.
For example, a user's name or post's text could be updated, and I may have now new users/posts that need to be created.

I was looking for `upsertMany` in the docs, but looks like it doesn't exist (yet?).

Here is what I'm doing at the moment:

```ts
await prisma.$transaction(
  users.map((user) =>
    prisma.user.upsert({
      where: { id: user.id },
      update: { name: user.name },
      create: user,
    })
  )
);

await prisma.$transaction(
  posts.map((post) =>
    prisma.post.upsert({
      where: { id: post.id },
      update: { text: post.text, userId: post.userId },
      create: post,
    })
  )
);
```

Is this the best I can do, or is there a more efficient/performant way to upsert users and posts in this situation?

**Comment by ryands17** [comment, marked as answer]
@moroshko 👋 

Currently yes this is the only way you could do it via Prisma's provided API. It would be great if you could add a 👍 to this request so that we can look at the priority for this.
https://github.com/prisma/prisma/issues/4134

**Reply by moroshko** [reply]
Thanks @ryands17 for confirming. Already thumbed up 👍

**Comment by obzenner** [comment]
👍 PLEASE <3

**Comment by 6peterlu** [comment]
🙏 🙏 🙏  This would be super helpful!

**Comment by dkstrong** [comment]
just ran in to this situation myself. would be a super useful way to do this common operation!

**Comment by pencilcheck** [comment]
Also running into this problem, upsert or update will just timeout even if you cram everything in a $transaction

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
